assets/js/testScript.js

```diff
@@ -0,0 +1,488 @@
+      fireDirection = 'right'
+    } // right
+    if (keys[67]) { fireBullet() } //default attack
+    if (keys[68]) { skillOne() } // purple wave (D)
+    if (keys[90]) { skillTwo() } // giant panda (X)
+    if (keys[88]) { skillThree() }// giant hammer(Z)
+    e.preventDefault() // prevent the default action (scroll / move caret)
+  })
+  $body.on('keyup', function (e) {    // stop player sprite animation when not moving or attacking and reset player Y-axis to counteract jump
+    keys[e.which] = false
+    var $player = $('.player')
+    var $playPos = $player.position()
+    $player.css('webkitAnimationPlayState', 'paused')
+    // console.log(` current position ${$playPos.top}`)
+    if ($playPos.top !== 590) { $player.css('top', `590px`) }
+  })
+
+  function bulletCollisionCheck (obj1, obj2) { //bullet collision check
+    var red = obj1.position()
+    var redX = red.left
+    var blue = obj2.jTarget.position()
+    var blueX = blue.left
+    var redY = red.top
+    var blueY = blue.top
+    if (blueX < redX + obj1.width() &&
+        blueX + obj2.jTarget.width() > redX &&
+        blueY < redX + obj1.height() &&
+        obj2.jTarget.height() + blueY > redY) {
+      obj2.hitPoints -= 1
+      $bullet.css('left', $playPos.left)
+      $bullet.css('top', '620px')
+      if (obj2.hitPoints === 0) {
+        obj2.jTarget.remove()
+        if (obj2.bosscheck === false) {
+          mobArray.splice(key, 1)
+          mobsOnScreen--
+          levelOneMobCount--
+        } else if (obj2.bosscheck) {
+          bossArray.splice(key, 1)
+          bossDead = true
+        }
+      }
+      if (mobsOnScreen === 0 && levelOneMobCount === 0) { //if mobs are dead, spawn boss
+        bossLevel = true
+      }
+      if (bossLevel === true && bossDead === true) {   //if boss is dead, end game
+        gameEnd = true
+      }
+      return true
+    } else { return false }
+  }
+
+  function playerMobCollisionCheck (obj1, obj2) {     //player - mob collision check
+    var red = obj1.position()
+    var redX = red.left
+    var blue = obj2.jTarget.position()
+    var blueX = blue.left
+    var redY = red.top
+    var blueY = blue.top
+
+    if (blueX < redX + $player.width() &&
+         blueX + obj2.jTarget.width() > redX &&
+         blueY < redX + $player.height() &&
+         obj2.jTarget.height() + blueY > redY) {
+      console.log('PLAYER HEALTH REDUCED')
+      playerHealth -= obj2.damage
+      var currentHpWidth = $('.hpBar').width()
+      $('.hpBar').css('width', `${currentHpWidth - obj2.damage / 5}px`)
+      $hpBar.text(`${playerHealth}/1000`)
+      if (playerHealth === 0) {
+        gameEnd = true
+      }
+    } else {
+      return false
+    }
+  }
+
+  function fireBullet () {      //bullet control functiom
+    $playPos = $player.position()
+    $mobPos = $mob.position()
+    $bullet.show()
+    $bulletLoc = $bullet.position()
+    $bulletExact = $bulletLoc.left
+    $bulletTop = $playPos.top
+    $bulletLeft = $playPos.left
```

Comment:
$bulletLeft is not used in any part of the script. Any reason for this var?

Reply:
nope, forgot to take out after refactoring code